lib/csv_generator.rb

```diff
@@ -0,0 +1,247 @@
+class CSVGenerator
+
+  # Let's declare these variables as read only functions. This way,
+  # when you declare `@output_file = opts[:out]` inside of initialize,
+  # users of that instance can still call a method to access that variable.
+  #
+  #    gen = CSVGenerator.new(opts)
+  #    puts "My Output file has #{gen.number_of_columns} columns in #{gen.output_file}"
+  #
+  # You'll also note that inside of your instance methods, you can
+  # replace `@output_file` with `output_file` simply because of this
+  # line.  Pretty cool, huh!
+  attr_reader :output_file, :number_of_columns, :number_of_rows
+
+  # So why the (!) bang? It will indicate to users that this is an
+  # actionable method, used to parse CLI arguments, similar to `OptionParser`
+  #
+  # By using this method, we can refactor it later on if we decided to
+  # use a proper parser.  For now, let's try ARGV, then prompt the user.
+  #
+  # This may look a little weird, but essentially we're deferring arg
+  # parsing validation to parse_argv, or if that returns nil, we call
+  # prompt_user.  It's essentially what you had earlier, but is a
+  # little more 'Ruby'.
+  #
+  # (ARGV.size != 3) ? prompt_user : parge_argv )
+  def self.parse!
+    new( parse_argv || prompt_user )
+  end
+
+  # Return nil if ARGV wasn't passed
+  def self.parse_argv
+    return if ARGV.size != 3
+    {out: ARGV[0], cols: ARGV[1], rows: ARGV[2]}
+  end
+
+  # We'll move up the instance method prompt_user and make some slight
+  # changes.  For now I'm removed the buffer method all together and
+  # just settled for newlines.
+  def self.prompt_user
+    args = {}
+    puts "Creation parameters can be passed to the script as arguments like so:"
+    puts "#{$0} <output_file> <number_of_columns> <number_of_rows>\n\n"
```

Comment:
Adding the newline here at the end of the string rather than calling a method seems plenty clean.